src/Core/Form/ChoiceProvider/CurrencyNameByIsoCodeChoiceProvider.php

```diff
@@ -61,7 +61,7 @@ public function getChoices()
             }
             $currencyNames = $cldrCurrency->getDisplayNames();
             $isoCode = $cldrCurrency->getIsoCode();
-            $displayName = sprintf('%s (%s)', $currencyNames['default'], $isoCode);
+            $displayName = sprintf('%s (%s)', $currencyNames['default'] ?? null, $isoCode);
```

Comment:
I suggest to add a new method here https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Localization/CLDR/CurrencyData.php `CurrencyData::getDefaultDisplayName():?string`